### PR TITLE
To fix #3.

### LINKED ISSUE #3: Stop watching position if detached is called
Do the clean up. Reuse https://github.com/ebidel/geo-location/blob/master/geo-location.html#L136

### PR DESCRIPTION
**Pull request by motss**
clearWatch when the element is detached from DOM.

### CODE REVIEW

**Comment by motss** [comment]
@ebidel PTAL.

File: geo-location.html [hunk 1]
```diff
@@ -142,16 +142,30 @@
...
+        if (watch_ !== null && watch_ >= 0) {
+          this._clearWatch(watch_);
+        }
+      },
+
+      /**
+       * Stop updating latitude/longitude as the device changes position.
+       * @param {Number} watch_ watch ID value.
+       */
+      _clearWatch: function (watch_) {
```
Comment (ebidel) [inline geo-location.html]: watch_ -> watch

File: geo-location.html [hunk 2]
```diff
@@ -142,16 +142,30 @@
...
 
+      detached: function () {
+        if (watch_ !== null && watch_ >= 0) {
+          this._clearWatch(watch_);
+        }
+      },
+
+      /**
+       * Stop updating latitude/longitude as the device changes position.
+       * @param {Number} watch_ watch ID value.
```
Comment (ebidel) [inline geo-location.html]: watch_ -> watch

File: geo-location.html [hunk 3]
```diff
@@ -168,8 +182,7 @@
 
         if (this.watchPos) {
           watch_ = navigator.geolocation.watchPosition(success, error, options);
-        }
-        else {
+        }else {
```
Comment (ebidel) [inline geo-location.html]: `} else {`

File: geo-location.html [hunk 4]
```diff
@@ -142,16 +142,30 @@
        *   @param {Number} detail.longitude Longitude of the current position.
        */
 
+      detached: function () {
+        if (watch_ !== null && watch_ >= 0) {
```
Comment (ebidel) [inline geo-location.html]: i dont think we need the `watch_ >= 0` check

File: geo-location.html [hunk 5]
```diff
@@ -142,16 +142,30 @@
...
         this._setPosition(null);
         this._setLatitude(null);
         this._setLongitude(null);
       },
 
       fetch: function () {
-        if (watch_) {
-          navigator.geolocation.clearWatch(watch_);
-          watch_ = null;
+        if (watch_ !== null && watch_ >= 0) {
```
Comment (ebidel) [inline geo-location.html]: dont' need `watch_ >= 0`

**Comment by ebidel** [comment]
Reviewed

**Comment by motss** [comment]
@ebidel PTAL.

File: geo-location.html [hunk 6]
```diff
@@ -142,16 +142,30 @@
...
+        }
+      },
+
+      /**
+       * Stop updating latitude/longitude as the device changes position.
+       * @param {Number} watch watch ID value.
+       */
+      _clearWatch: function (watch) {
+        navigator.geolocation.clearWatch(watch);
+        watch = null;
```
Comment (ebidel) [inline geo-location.html]: dont need this

**Comment by motss** [comment]
@ebidel PTAL.

File: geo-location.html [hunk 7]
```diff
@@ -142,16 +142,30 @@
        *   @param {Number} detail.longitude Longitude of the current position.
        */
 
+      detached: function () {
+        if (watch_ !== null) {
```
Comment (ebidel) [inline geo-location.html]: Sorry one last thing. Now that I think about it, we can probably just moved the null check into the functional and have calling locations just call it with checking.

Comment (motss) [inline geo-location.html]: Move null check into `_clearWatch`?

File: geo-location.html [hunk 8]
```diff
@@ -142,16 +142,30 @@
...
         this._setPosition(null);
         this._setLatitude(null);
         this._setLongitude(null);
       },
 
       fetch: function () {
-        if (watch_) {
-          navigator.geolocation.clearWatch(watch_);
-          watch_ = null;
+        if (watch_ !== null) {
```
Comment (motss) [inline geo-location.html]: How about this? Do we actually need to clear existing watch everytime we run `fetch`?

File: geo-location.html (hunk 7 above)
Comment (ebidel) [inline geo-location.html]: Correct

File: geo-location.html (hunk 8 above)
Comment (ebidel) [inline geo-location.html]: I would imagine that's a good thing? You don't want a callback to fire for an old watch if you've just started a new one.

**Comment by motss** [comment]
@ebidel PTAL at latest commit.

**Comment by ebidel** [comment]
Lgtm

**Pull request merged**